v0/util/errors/base.js

```diff
@@ -13,31 +15,21 @@ class RudderErrorBase extends Error {
   }
 
   static getStatTags(statTags, defaults) {
-    let finalStatTags = statTags;
-    const { defScope, defMeta, destination } = defaults;
-    if (!finalStatTags || Array.isArray(finalStatTags)) {
-      finalStatTags = {
-        scope: defScope,
-        meta: defMeta
-      };
-    }
-
-    const tagNames = Object.keys(finalStatTags);
-    if (!tagNames.includes("scope")) {
-      finalStatTags = {
-        ...finalStatTags,
-        scope: defScope
-      };
-    }
+    if (isObject(statTags)) {
+      const finalStatTags = Object.keys(statTags)
+        .filter(tagKey => ["scope", "meta", "destType"].includes(tagKey))
```

Comment:
This is not required.

Reply:
Why do you think the filtering would not be required here ?